ansible_rulebook/cli.py

```diff
@@ -46,16 +46,21 @@
 import sys
 from typing import List, NoReturn
 
-from ansible_rulebook.job_template_runner import job_template_runner
+import ansible_rulebook.util as util
 
-if os.environ.get("JAVA_HOME") is None:
-    print("JAVA_HOME is not set. " "Please install Java 11+ and set JAVA_HOME")
-    sys.exit(1)
+# ensure a valid JVM is available and configures JAVA_HOME if necessary
+# must be done before importing any other modules
+util.check_jvm()
+if not os.environ.get("JAVA_HOME"):
+    os.environ["JAVA_HOME"] = util.get_java_home()
```

Comment:
Is there a reason why this check must be performed before other imports?

Reply:
The other imports load the drools stuff, in that point, you need the JVM

Reply:
I'm struggling to find it. Can you please point where the drools stuff is loaded?

Reply:
I think is https://github.com/ansible/ansible-rulebook/blob/main/ansible_rulebook/engine.py#L24 -> https://github.com/ansible/drools_jpy/blob/main/src/drools/ruleset.py

Correct me if I'm wrong @mkanoor 

This was already discovered some time ago, I am not adding it now.